Server/api/routes/referee/matchups.ts

```diff
@@ -0,0 +1,351 @@
+import Router from "@koa/router";
+import { TournamentRoleType, unallowedToPlay } from "../../../../Interfaces/tournament";
+import { Matchup } from "../../../../Models/tournaments/matchup";
+import { Matchup as MatchupInterface } from "../../../../Interfaces/matchup"; 
+import { Team as TeamInterface } from "../../../../Interfaces/team";
+import { Team } from "../../../../Models/tournaments/team";
+import { TournamentRole } from "../../../../Models/tournaments/tournamentRole";
+import { Mappool } from "../../../../Interfaces/mappool";
+import { discordClient } from "../../../discord";
+import { isLoggedInDiscord } from "../../../middleware";
+import { hasRoles, validateTournament } from "../../../middleware/tournament";
+
+// TODO: Delete these after testing
+import { Stage } from "../../../../Models/tournaments/stage";
+import { Tournament } from "../../../../Models/tournaments/tournament";
+import { OAuth, User } from "../../../../Models/user";
+import { Round } from "../../../../Models/tournaments/round";
+
+const refereeMatchupsRouter = new Router();
+
+//TODO: Look into making refereeRouter.use work for the middleware functions
+function convertTeam(team: Team): TeamInterface;
+function convertTeam(team: null | undefined): undefined;
+function convertTeam (team: Team | null | undefined): TeamInterface | undefined
+function convertTeam (team: Team | null | undefined): TeamInterface | undefined {
+    return team ? {
+        ID: team.ID,
+        name: team.name,
+        abbreviation: team.abbreviation,
+        timezoneOffset: team.timezoneOffset,
+        avatarURL: team.avatarURL,
+        pp: team.pp,
+        rank: team.rank,
+        BWS: team.BWS,
+        manager: {
+            ID: team.manager.ID,
+            username: team.manager.osu.username,
+            osuID: team.manager.osu.userID,
+            BWS: 0,
+            isManager: true,
+        },
+        members: team.members.map(member => ({
+            ID: member.ID,
+            username: member.osu.username,
+            osuID: member.osu.userID,
+            BWS: 0,
+            isManager: member.ID === team.manager.ID,
+        })),
+    } : undefined;
+}
+
+refereeMatchupsRouter.get("/:tournamentID", validateTournament, isLoggedInDiscord, hasRoles([TournamentRoleType.Organizer, TournamentRoleType.Referees]), async (ctx) => {
```

Comment:
this is going to be the heaviest endpoint of all times without pagination. i'd expect it to already churn pretty hard, even at this stage of CO.